app/main.py

```diff
@@ -1,3 +1,63 @@
+from typing import Union
```

Comment:
Add `from __future__ import annotations` here for class annotation with `Distance` to work.